suite-native/atoms/src/TouchableSwitchRow.tsx

```diff
@@ -0,0 +1,66 @@
+import { ReactNode } from 'react';
+import { TouchableOpacity } from 'react-native';
+
+import { Icon, IconName } from '@suite-native/icons';
+
+import { Box } from './Box';
+import { Card } from './Card/Card';
+import { HStack, VStack } from './Stack';
+import { Switch } from './Switch';
+import { Text } from './Text';
+
+type TouchableSwitchRowProps = {
+    isChecked: boolean;
+    onChange: (value: boolean) => void;
+    text: ReactNode;
+    accessibilityLabel: string;
+    description?: ReactNode;
+    iconName: IconName;
+    testID?: string;
+};
+
+export const TouchableSwitchRowDescription = ({ children }: { children: ReactNode }) => (
+    <Text variant="hint" color="textSubdued">
+        {children}
+    </Text>
+);
+
+export const TouchableSwitchRow = ({
+    isChecked,
+    onChange,
+    text,
+    description,
+    accessibilityLabel,
+    iconName,
+    testID,
+}: TouchableSwitchRowProps) => {
+    const handleChange = () => {
+        onChange(!isChecked);
+    };
+
+    return (
+        <Card borderColor="borderElevation1" noPadding>
+            <TouchableOpacity
+                onPress={handleChange}
+                accessibilityLabel={accessibilityLabel}
+                accessibilityRole="switch"
+                accessibilityState={{ checked: isChecked }}
+            >
+                <HStack margin="sp16" spacing="sp12">
+                    <Box marginVertical="sp2">
+                        <Icon name={iconName} size="mediumLarge" />
+                    </Box>
+                    <HStack justifyContent="space-between" flex={1}>
+                        <VStack flex={1} spacing="sp2">
+                            <Text variant="highlight">{text}</Text>
+                            <Text variant="hint" color="textSubdued">
+                                {description}
+                            </Text>
+                        </VStack>
+                        <Switch testID={testID} isChecked={isChecked} onChange={handleChange} />
```

Comment:
Since `onChange` is a component property, I'd use it here directly:
```suggestion
                        <Switch testID={testID} isChecked={isChecked} onChange={() => onChange(!isChecked)} />
```

Reply:
It's used in the TouchableOpacity as well so you can click on the whole Card, so I'd prefer to keep it like this. That's how it was before too (with the onChange mounted on switch as well as touchable opacity)

Reply:
I don't think it was used there before. Or I must have missed it. 🤔 If it's used on two places then it of course makes sense to reuse the function.